pkg/manager/member/pd_upgrader.go

```diff
@@ -109,25 +109,16 @@ func (u *pdUpgrader) upgradePDPod(tc *v1alpha1.TidbCluster, ordinal int32, newSe
 	tcName := tc.GetName()
 	upgradePdName := PdName(tcName, ordinal, tc.Namespace, tc.Spec.ClusterDomain)
 	upgradePodName := PdPodName(tcName, ordinal)
+
+	// If current pd is leader, transfer leader to other pd
 	if tc.Status.PD.Leader.Name == upgradePdName || tc.Status.PD.Leader.Name == upgradePodName {
 		var targetName string
 		if tc.PDStsActualReplicas() > 1 {
-			targetOrdinal := helper.GetMaxPodOrdinal(*newSet.Spec.Replicas, newSet)
-			if ordinal == targetOrdinal {
-				targetOrdinal = helper.GetMinPodOrdinal(*newSet.Spec.Replicas, newSet)
-			}
-			targetName = PdName(tcName, targetOrdinal, tc.Namespace, tc.Spec.ClusterDomain)
-			if _, exist := tc.Status.PD.Members[targetName]; !exist {
-				targetName = PdPodName(tcName, targetOrdinal)
-			}
+			targetName = choosePDToTransferFromMembers(tc, newSet, ordinal)
```

Comment:
Should we also check the peer members if no member is available?

Reply:
LTGM, can we merge the functions `choosePDToTransferFromPeerMembers` and `choosePDToTransferFromPeerMembers` ?

Reply:
1. Find the max suitable ordinal in (x, n], because they have been upgraded
2. If no suitable ordinal, find the min suitable ordinal in [0, x) to reduce the count of transfer
3. If no suitable ordinal, find the peer member

Reply:
Done